core/src/main/scala/org/ensime/config/package.scala

```diff
@@ -10,12 +10,19 @@ import org.ensime.util.file._
 package object config {
 
   implicit class RichEnsimeConfig(val c: EnsimeConfig) extends AnyVal {
-    def scalaSourceFiles: Set[File] = for {
-      module: EnsimeModule <- c.modules.values.toSet
-      root <- module.sourceRoots
-      file <- root.tree
-      if file.isFile && file.getName.endsWith(".scala")
-    } yield file
+    def scalaSourceFiles: Set[File] =
+      c.modules.values.toSet.flatMap((m: EnsimeModule) => m.scalaSourceFiles)
```

Comment:
you might be able to use `breakOut` here instead of having to call explicit `.toSet`